### PR TITLE
Restructure files

### LINKED ISSUE #52: Remove unused code from container
### Description

Currently, we are zipping the whole src folder and uploading it to the container.
The code in the `frontend/` folder is not needed after we compile the `static` folder.

https://github.com/Azure-Samples/rag-postgres-openai-python/blob/b05f38ab45698331561b3c252e2a2c5c7da7bae9/azure.yaml#L7

### Proposal

**[Option 1]** Move the `frontend` folder outside of the `src` folder

**[Option 2]** Restructure `src` folder to this:

```md
src
│
└─── frontend
│    .....
└─── backend
    │    .dockerignore
    │    Dockerfile
    │    .....
    └─── fastapi_app
         │    .....
         └─── static
```

Then use backend folder in `azure.yaml`

### PR DESCRIPTION
**Pull request by john0isaac**
## Purpose

Fix #52 

## Does this introduce a breaking change?

When developers merge from main and run the server, azd up, or azd deploy, will this produce an error?
If you're not sure, try it out on an old environment.

```
[ ] Yes
[x] No
```

## Type of change

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Code quality checklist

See [CONTRIBUTING.md](https://github.com/Azure-Samples/rag-postgres-openai-python/blob/main/CONTRIBUTING.md#submit-pr) for more details.

- [-] The current tests all pass (`python -m pytest`).
- [-] I added tests that prove my fix is effective or that my feature works
- [-] I ran `python -m pytest --cov` to verify 100% coverage of added lines
- [-] I ran `python -m mypy` to check for type errors
- [-] I either used the pre-commit hooks or ran `ruff` manually on my code.

### CODE REVIEW

File: src/pyproject.toml
Comment (john0isaac) [inline src/pyproject.toml:1]: This is a workaround to keep using pyproject.toml.

The only down side is that now (locally) you'll need to install the fastapi_app package before installing the backend package as the backend package can't resolve the fastapi_app source, I tried specifying the path but it needs a full path which will be a nightmare on different platforms.
https://stackoverflow.com/questions/75159453/specifying-local-relative-dependency-in-pyproject-toml

File: README.md [hunk 1]
```diff
@@ -128,9 +128,10 @@ Since the local app uses OpenAI models, you should first deploy it for the optim
 1. Run these commands to install the web app as a local package (named `fastapi_app`), set up the local database, and seed it with test data:
 
     ```bash
+    python3 -m pip install -e src/backend
     python3 -m pip install -e src
-    python ./src/fastapi_app/setup_postgres_database.py
-    python ./src/fastapi_app/setup_postgres_seeddata.py
+    python ./src/backend/fastapi_app/setup_postgres_database.py
+    python ./src/backend/fastapi_app/setup_postgres_seeddata.py
```
Comment (john0isaac) [inline README.md:134]: This now how it works but it's not pretty due to the issue I mentioned below.

vscode resolution will still work as is and running the launch.json scripts will work as is.

File: src/pyproject.toml
Comment (john0isaac) [reply src/pyproject.toml:1]: Btw we can use a package manager (poetry or smth) and we will be able to specify a relative path but I know you are not a big fan of this approach so, I didn't do it. But it's an option too.

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -54,7 +54,7 @@
 		}
 	},
 	// Use 'postCreateCommand' to run commands after the container is created.
-	"postCreateCommand": "pip install -r requirements-dev.txt && pip install -e src",
+	"postCreateCommand": "pip install -r requirements-dev.txt && pip install -e src/backend && pip install -e src",
```
Comment (pamelafox) [inline .devcontainer/devcontainer.json:57]: It feels weird that we have to install both. What happens if we only install src/backend?

Comment (john0isaac) [reply .devcontainer/devcontainer.json]: nothing, guess I never tried it lol. it works 🤣

**Comment by pamelafox** [comment]
mypy is unhappy (as usual). I'm placating it.

**Comment by john0isaac** [comment]
What's weird is that these tests passed last Thursday.
I upgraded my local mypy version and got the same errors. I think it's the latest release.
Thanks! For fixing it.

**Comment by pamelafox** [comment]
@john0isaac Yep latest mypy came out last week, and caused a bunch of issues all over. It broke azure-search-openai-demo CI cuz of https://github.com/python/mypy/issues/17550 which I've also just fixed.

**Pull request merged**